hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/UpdateProcessorSupplier.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.sql.impl.connector.mongodb;
+
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.mongodb.WriteMode;
+import com.hazelcast.jet.mongodb.impl.WriteMongoP;
+import com.hazelcast.jet.mongodb.impl.WriteMongoParams;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.row.JetSqlRow;
+import com.hazelcast.sql.impl.schema.Table;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.UpdateOneModel;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.model.WriteModel;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.bson.types.ObjectId;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static com.hazelcast.jet.mongodb.MongoDBSinkBuilder.DEFAULT_COMMIT_RETRY_STRATEGY;
+import static com.hazelcast.jet.mongodb.MongoDBSinkBuilder.DEFAULT_TRANSACTION_OPTION;
+import static com.hazelcast.jet.mongodb.impl.Mappers.defaultCodecRegistry;
+import static java.util.Arrays.asList;
+
+/**
+ * ProcessorSupplier that creates {@linkplain WriteMongoP} processors on each instance,
+ *  that will update given items.
+ */
+public class UpdateProcessorSupplier implements ProcessorSupplier {
+
+    private final String connectionString;
+    private final String databaseName;
+    private final String collectionName;
+    private final List<String> fieldNames;
+    private final List<Object> updates;
```

Comment:
updates field does not indicates that it is serializable. 
Maybe make it something like
private final List<? extends Serializable> updates;